### PR TITLE
Upgrade to Sphinx 1.4.9

### PR DESCRIPTION
**Pull request by jayvdb**
Use 'compatible with' (~=) requirements syntax.

Closes https://github.com/coala/coala/issues/3200

### CODE REVIEW

**Comment by Mixih** [comment]
it's weird, docs is throwing error, circle just doesn't see it:
From circle docs log:
>Warning, treated as error:
/home/ubuntu/coala/docs/Developers/Writing_Tests.rst:136: WARNING: unknown document: Testing

**Comment by Mixih** [comment]
unack 688a25b

**Comment by jayvdb** [comment]
Why is that relevant?
It was failing before, and is failing now.
This isnt about fixing the docs.

File: docs-requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # all requirements are needed so that autodoc can find them
 -r requirements.txt
 -r test-requirements.txt
-sphinx==1.4.8
+sphinx~=1.4
```
Comment (sils) [inline docs-requirements.txt]: why not `sphinx~=1.4.9`?

Comment (jayvdb) [inline docs-requirements.txt]: I'm happy to do that, if preferred.

Comment (sils) [inline docs-requirements.txt]: let's go for it IMO

**Comment by sils** [comment]
imported this to https://github.com/coala/coala/pull/3204

**Pull request closed without merging**